server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -186,7 +193,25 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
     final boolean isQueryEndpoint = request.getRequestURI().startsWith("/druid/v2")
                                     && !request.getRequestURI().startsWith("/druid/v2/sql");
 
-    if (isQueryEndpoint && HttpMethod.DELETE.is(request.getMethod())) {
+    final boolean isAvatica = request.getRequestURI().startsWith("/druid/v2/sql/avatica");
+
+    if (isAvatica) {
+      byte[] requestContent = getRequestContent(request);
+      String connectionId = getAvaticaConnectionId(requestContent, objectMapper);
+      Server targetServer = avaticaConnectionBalancer.balance(hostFinder.getAllServers(), connectionId);
+      if (targetServer == null) {
```

Comment:
Suggest moving this logic into a `QueryHostFinder` method named something like `findServerAvatica(String connectionId)`, and handling the no-brokers-found situation more similarly to how that class handles it for native Druid queries.

Reply:
Moved this into QueryHostFinder